### PR TITLE
Featureimage/ipmprog 1481 handle port unavailability

### PR DESCRIPTION
**Pull request by FrancoisRegisDegott-eaton**

### CODE REVIEW

File: server/src/jobs/impl/neon.cpp [hunk 1]
```diff
@@ -22,11 +22,19 @@
...
+static int verify_fn(void* /*userdata*/, int /*failures*/, const ne_ssl_certificate* /*cert*/)
+{
+    return 0; // trust cert
+}
+
 namespace neon {
 
-Neon::Neon(const std::string& address, uint16_t port, uint16_t timeout)
-    : m_session(ne_session_create("http", address.c_str(), port), &closeSession)
+Neon::Neon(const std::string& scheme, const std::string& address, uint16_t port, uint16_t timeout)
```
Comment (perrettecl) [inline server/src/jobs/impl/neon.cpp:32]: name is not consistant with interface

**Review by perrettecl — APPROVED** [review]

File: server/src/jobs/impl/neon.cpp (hunk 1 above)
Comment (FrancoisRegisDegott-eaton) [reply server/src/jobs/impl/neon.cpp:32]: done

File: server/src/jobs/protocols.cpp [hunk 2]
```diff
@@ -34,87 +34,112 @@ namespace fty::job {
 enum class Type
 {
     Powercom = 1,
-    Xml      = 2,
-    Snmp     = 3,
+    Snmp     = 2,
```
Comment (ericclappier-eaton) [inline server/src/jobs/protocols.cpp]: I think xml must be prioritized than snmp

File: server/src/jobs/protocols.cpp [hunk 3]
```diff
@@ -34,87 +34,112 @@ namespace fty::job {
...
-        switch (prot) {
-            case Type::Snmp:
-                out.append("nut_snmp");
+    // supported protocols, with defaults
+    struct {
+        Type protocol;
+        std::string protocolStr;
+        uint16_t port;
+    } tries[] = {
+        {Type::Powercom, "nut_powercom", 443},
```
Comment (ericclappier-eaton) [inline server/src/jobs/protocols.cpp:89]: I think xml must be prioritized than snmp

File: server/src/jobs/protocols.cpp (hunk 2 above)
Comment (FrancoisRegisDegott-eaton) [reply server/src/jobs/protocols.cpp]: there is no (more) priority between protocols... we only check if each protocol is reachable on its default port...

File: server/src/jobs/protocols.cpp (hunk 3 above)
Comment (FrancoisRegisDegott-eaton) [reply server/src/jobs/protocols.cpp:89]: done. the preference order is: powercom, xml, snmp

**Review by ericclappier-eaton — APPROVED** [review]

**Pull request closed without merging**